### PR TITLE
test: Added tests for rhc status

### PR DESCRIPTION
**Pull request by Archana-PandeyM**
One test currently fails due to known issue :https://issues.redhat.com/browse/CCT-525

### CODE REVIEW

File: integration-tests/utils/__init__.py
Comment (jirihnidek) [inline integration-tests/utils/__init__.py:15]: I would add some doc string here. It is not clear what this function returns.

File: integration-tests/test_status.py
Comment (jirihnidek) [inline integration-tests/test_status.py:75]: Well, if it is not possible to test starting yggdrasil service after reboot ATM, shouldn't we rename this function and add proper function for this test case (starting of yggdrasil service after reboot), when it will be possible?

You know. It is confusing. When you look at the list of methods, then it seems that something is tested, but it is not true. You have to read all comments to realize that this function is a fake test.

**Review by jirihnidek — CHANGES_REQUESTED** [review]
Thanks for PR. I have few comments and proposals.

File: integration-tests/utils/__init__.py
Comment (Lorquas) [inline integration-tests/utils/__init__.py:19]: the `sh` library can throw additional tracebacks depending on the systemctl's exit code, which it likely will if this is ever run (this code only happens in exceptions).  To make debugging cleaner would this benefit from either exception handling, or even using `subprocess` - maybe even pytest_client_tools'  `logged_run` ?

File: integration-tests/test_status.py
Comment (Lorquas) [inline integration-tests/test_status.py:54]: This assert is failing in the downstream and upstream tests?

File: integration-tests/test_status.py
Comment (Lorquas) [inline integration-tests/test_status.py:70]: is this effectively the same as `rhc.run("disconnect", check=False)` which is used in the previous test above?

**Review by Lorquas — COMMENTED** [review]
Generally good!  I have a few questions (see inline)...

File: integration-tests/test_status.py
Comment (Archana-PandeyM) [reply integration-tests/test_status.py:54]: yes, because of known issue https://issues.redhat.com/browse/CCT-525

File: integration-tests/utils/__init__.py
Comment (Archana-PandeyM) [reply integration-tests/utils/__init__.py:19]: make sense to use existing logged_run.Thanks for feedback!

**Review by Lorquas — APPROVED** [review]
Approved pending discussion on cct-525

File: integration-tests/utils/__init__.py
Comment (ptoscano) [reply integration-tests/utils/__init__.py]: well no, `logged_run()` is currently NOT public API (even if it isn't named with `_`)

**Comment by jirihnidek** [comment]
It seems that merging this PR is still blocked by some issue in insights-client on Fedora.

**Comment by Lorquas** [comment]
> It seems that merging this PR is still blocked by some issue in insights-client on Fedora.

insights-client is having some non-related problems on latest python: https://issues.redhat.com/browse/RHINENG-11283

**Comment by jirihnidek** [comment]
> > It seems that merging this PR is still blocked by some issue in insights-client on Fedora.
> 
> insights-client is having some non-related problems on latest python: https://issues.redhat.com/browse/RHINENG-11283

OK, then it would be good to skip failing test on Fedora ATM using e.g. `@pytest.mark.skipif(platform.freedesktop_os_release()["ID"] == "fedora", reason="RHINENG-11283")` and create another card to enable these test, when related issue with insights-client is fixed. It is not good practice to approve failed tests.

**Comment by Archana-PandeyM** [comment]
@jirihnidek the problem with fedora build is resolved with insights-core fix! https://github.com/RedHatInsights/insights-core/pull/4154

File: integration-tests/test_status.py
Comment (jirihnidek) [inline integration-tests/test_status.py:65]: This is actually test of yggdrasil and I'm not sure if it is good to have it here, because when yggdrasil could not be restarted on e.g. disconnected system, then we will not be able to fix it in this repository.

**Review by jirihnidek — APPROVED** [review]
LGTM, I would also add some test for `--format json`, but we can add it in the future in another PR.

**Comment by Archana-PandeyM** [comment]
> LGTM, I would also add some test for `--format json`, but we can add it in the future in another PR.
yes we will increase the test coverage to have this tests in another PR.

**Pull request merged**
